website/blog/2022-09-28-analyst-to-ae.md

```diff
@@ -0,0 +1,178 @@
+
+The first time that a team of Analysts and Data Engineers build a curated dataset, they will often expect this process to be a straight line to completion. Expectations are that the process will look something like this:
+
+![The expectation](/img/blog/2022-09-28-analyst-to-ae/first_loop.png)
+
+What quickly becomes apparent is that this push for a linear process often ends up creating three unanticipated “loops”:
+
+![The reality](/img/blog/2022-09-28-analyst-to-ae/multiple_loops.png)
+
+## Loop 1: Reacting to findings in the raw data
+
+Let’s say that your Analyst has defined the business need as an `Active Users` dataset with “Unique users that login during a given day”. The Analyst will try to do as much discovery work up-front as possible, because it’s hard to predict exactly what you’ll find in the raw data. When a Data Engineer gets stuck writing a model, they’ll need to reach out to the Analyst for some additional discovery. When an Analyst-turned-Analytics-Engineer has a question while writing a model, they don’t need to wait to talk to anyone, and they can begin researching it immediately. Which brings us to our first point:
+
+<h3 align="center">Analysts already know the data that they want.</h3>
+
+If the raw `Login` dataset contains two different date (`Login_Date` and `Session_Date`) fields, a Data Engineer would be stuck. They can’t just guess because using the wrong date field will create an entirely different metric! So they must go back to the Analyst for clarity on which field to use. We’ve just had a full loop with two hand-offs, and the Data Engineer hasn’t even begun building the model yet.
+
+| Login_Date | Session_Date | User_Id |
+|------------|--------------|---------|
+| 2022-08-01 | 2022-08-01   | 123     |
+| 2022-08-01 | 2022-08-03   | 123     |
+| 2022-08-01 | 2022-08-04   | 123     |
+| 2022-08-01 | 2022-08-01   | 427     |
+| 2022-08-02 | 2022-08-02   | 864     |
+| 2022-08-03 | 2022-08-03   | 864     |
+| 2022-08-04 | 2022-08-04   | 975     |
+| 2022-08-04 | 2022-08-04   | NULL    |
+
+In comparison, your new Analytics Engineer could quickly pivot based on the findings in the raw data. They can skip the endless loops of “discovery”, because they can react in-the-moment. When they see the two date fields for `Login_Date` and `Session_Date`, they can instantly apply their product knowledge, compare against examples, or reach out to their business contacts to diagnose the difference and select the correct date for the model.
+
+If the business need includes looking at `Active Users` by `Groups A, B, and C`, then that adds an extra layer of complexity. Without an Analytics Engineer, you’ll see extra loops and hand-offs to finalize all the business logic, how to handle NULL values, and even just the final format.
+
+### Model format returned by data engineering
```

Comment:
This section is my least favorite.  I don't like these two being "sections" and then the tables feel _super_ long when they aren't very important at all.  Maybe we can switch to a bold "Model Format Option A" and a bold "Model Format Option B"?  For the first table, maybe only 1 header row that says `Date`, `Active Users Group A`, `Active Users Group B`, `Active Users Group C`?  And then reduce the rows to like 78-80.  And then the second table would be `Date`, `Group`, `Active Users`, and then only like rows 91-93

Reply:
Ok I reformatted the tables (it's a bummer we can't do side-by-side comparison), but I do feel that the existing headers (Format returned by data engineering vs Format needed for the BI tool) are more descriptive for the reader!